docs/public/specifications/tests/Address-TP/address01.md

```diff
@@ -1,47 +1,147 @@
+   *Name Server IP* set. 
 
-### Special procedural requirements
+3. Obtain the IP addresses of each name server for the domain using the method 
+   [Get-Zone-NS-Names-and-IPs] and add any non-duplicate results to 
+   *Name Server IP* set. 
 
-The registries listed in bullet 2 above must be fetched prior to testing 
+4. Create the following empty sets:
+   1. Name server name and IP address ("Documentation Address").
+   2. Name server name and IP address ("Local Use Address").
+   3. Name server name and IP address ("Not Globally Reachable").
 
-### Intercase dependencies
+5. For each name server in *Name Server IP* do:
+   1. Match the IP address against the IP ranges specified in 
+      [Special purpose IPv4 addresses] and [Special purpose IPv6 addresses]
+      1. If if falls within any of the address ranges reserved for 
+        *Documentation*, add the name server name and IP address to the
+        *Documentation Address* set,
+      2. Else, if it falls within an address range belonging to any of the 
+         following categories:  
+         - *Private-Use (IPv4)*
+         - *Loopback (IPv4)*
+         - *Loopback Address (IPv6)*
+         - *Link Local (IPv4)*
+         - *Link-Local Unicast* (IPv6)
+         - *Unique-Local* (IPv6)
+         - *Shared Address Space* (IPv6)
+         add the name server name and IP address to the *Local Use Adddress* 
+         set. 
+      3. Else, if the IP falls within a range that is not registered as 
+         *Globally Reachable*, add the name server name and IP address to 
+         the *Not Globally Reachable* set.
+   2. Go to the next server.
+6. If the sets *Documentation Address*, *Local Use Adddress* and 
+   *Not Globally Reachable* are all empty, then output 
+   *[A01_ADDR_GLOBALLY_REACHABLE]*
+7. Else, if the union of the *Documentation Address*, *Local Use Address* and 
+   *Not Globally Reachable* sets is equal to the *Name Server IP* set,
+   then output *[A01_NO_GLOBALLY_REACHABLE_ADDR]* 
+8. Else do:
+   1. If the *Documentation Address* set is non-empty, then output 
+      *[A01_DOCUMENTATION_ADDR]* with a list of name server names and IP addresses
+      from the set.
+   2. If the *Local Use Address* set is non-empty, then output 
+      *[A01_LOCAL_USE_ADDR]* with a list of name server names and IP addresses
+      from the set.
+   3. If the *Not Globally Reachable* set is non-empty, then output 
+      *[A01_ADDR_NOT_GLOBALLY_REACHABLE]* with a list of name server names and 
+      IP addresses from the set.
+
+  
+## Outcome(s)
+
+The outcome of this Test Case is "fail" if there is at least one message
+with the severity level *[CRITICAL]*.
+
+The outcome of this Test Case is "warning" if there is at least one message
+with the severity level *[ERROR]*, but no message with severity level
+*[CRITICAL]*.
+
+In other cases, no message or only messages with severity level
+*[INFO]*  the outcome of this Test Case is "pass".
```

Comment:
The language in [Basic01] says
```
The outcome of this Test Case is "fail" if there is at least one message with the severity
level ERROR or CRITICAL.

The outcome of this Test Case is "warning" if there is at least one message with the 
severity level WARNING, but no message with severity level ERROR or CRITICAL.

In other cases, no message or only messages with severity level INFO or NOTICE, 
the outcome of this Test Case is "pass".
```
[Basic01]: https://github.com/zonemaster/zonemaster/blob/master/docs/public/specifications/tests/Basic-TP/basic01.md#outcomes